### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Dmytro-Litinskyi**
[DEMO LINK](https://Dmytro-Litinskyi.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

**Review by tetiana-ivanchenko — CHANGES_REQUESTED** [review]
now i see error before loading todo

File: src/api.ts [hunk 1]
```diff
@@ -25,3 +26,29 @@ function get<T>(url: string): Promise<T> {
...
 
 export const getUser = (userId: number) => get<User>(`/users/${userId}`);
+
+// eslint-disable-next-line max-len
+export const getvisibleTodos = (
+  todos: Todo[],
+  query: string,
+  filter:FilterBy,
+) => {
+  let todosToFilter = todos;
```
Comment (mykhalenych) [inline src/api.ts:36]: better to use return instead of this

File: src/api.ts [hunk 2]
```diff
@@ -25,3 +26,29 @@ function get<T>(url: string): Promise<T> {
...
+
+  switch (filter) {
+    case FilterBy.Active:
+      todosToFilter = todos.filter(todo => !todo.completed);
```
Comment (mykhalenych) [inline src/api.ts:40]:
```suggestion
      return todos.filter(todo => !todo.completed);
```

File: src/components/TodoFilter/TodoFilter.tsx [hunk 3]
```diff
@@ -1,34 +1,82 @@
...
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            onChange={handleSelect}
+          >
+            {Object.values(FilterBy).map(current => (
+              <option value={current}>
```
Comment (mykhalenych) [inline src/components/TodoFilter/TodoFilter.tsx:47]: where is key?

**Pull request closed without merging**